map_model/src/intersection.rs

```diff
@@ -60,17 +60,16 @@ impl Intersection {
         self.intersection_type == IntersectionType::TrafficSignal
     }
 
-    pub fn get_incoming_lanes(&self, map: &Map, constraints: PathConstraints) -> Vec<LaneID> {
+    pub fn get_incoming_lanes<'slf>(&'slf self, map: &'slf Map, constraints: PathConstraints) -> impl Iterator<Item=LaneID> + 'slf {
```

Comment:
For consistency with the rest of the code, if there's only one explicit lifetime param, how about calling it `'a`?